### PR TITLE
Problem: not moving to third party folder (fix #130)

### LINKED ISSUE #130: Problem: not moving to third party folder
move qrcode*, defi-wallet-core* to third party folder
add license notice to source files
check with marketplace guideline

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: third party folder refactoring
add license
correct copy
windows ok
add windows library
reformat
update windows

### CODE REVIEW

File: Source/ThirdParty/PlayQrcodeLibrary/PlayQrcodeLibrary.Build.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (tomtau) [inline Source/ThirdParty/PlayQrcodeLibrary/PlayQrcodeLibrary.Build.cs:1]: missing the copyright headers?

**Review by tomtau — COMMENTED** [review]
likely ok, but maybe all the files need copyright and licensing header info (not sure of Epic's guidelines)?

File: Source/ThirdParty/PlayQrcodeLibrary/PlayQrcodeLibrary.Build.cs (hunk 1 above)
Comment (leejw51crypto) [reply Source/ThirdParty/PlayQrcodeLibrary/PlayQrcodeLibrary.Build.cs:1]: i'll check

Comment (leejw51crypto) [reply Source/ThirdParty/PlayQrcodeLibrary/PlayQrcodeLibrary.Build.cs]: added copyright header

**Comment by tomtau** [comment]
the build fails:
```
 [4/5] Link (lld) libUE4Editor-CronosPlayUnreal.so
81
  ld.lld: error: undefined symbol: qrcodegen::QrCode::encodeText(char const*, qrcodegen::QrCode::Ecc)
82
  >>> referenced by PlayCppSdkBPLibrary.cpp:172 (/home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Source/CronosPlayUnreal/Private/PlayCppSdkBPLibrary.cpp:172)
83
  >>>               /home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Intermediate/Build/Linux/B4D820EA/UE4Editor/Development/CronosPlayUnreal/Module.CronosPlayUnreal.cpp.o:(UPlayCppSdkBPLibrary::GenerateQrCode(FString))
84
  
85
  ld.lld: error: undefined symbol: qrcodegen::QrCode::getSize() const
86
  >>> referenced by PlayCppSdkBPLibrary.cpp:174 (/home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Source/CronosPlayUnreal/Private/PlayCppSdkBPLibrary.cpp:174)
87
  >>>               /home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Intermediate/Build/Linux/B4D820EA/UE4Editor/Development/CronosPlayUnreal/Module.CronosPlayUnreal.cpp.o:(UPlayCppSdkBPLibrary::GenerateQrCode(FString))
88
  
89
  ld.lld: error: undefined symbol: qrcodegen::QrCode::getModule(int, int) const
90
  >>> referenced by PlayCppSdkBPLibrary.cpp:181 (/home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Source/CronosPlayUnreal/Private/PlayCppSdkBPLibrary.cpp:181)
91
  >>>               /home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Intermediate/Build/Linux/B4D820EA/UE4Editor/Development/CronosPlayUnreal/Module.CronosPlayUnreal.cpp.o:(UPlayCppSdkBPLibrary::GenerateQrCode(FString))
92
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by leejw51crypto** [comment]
yes,  added linux build for qrcode

**Comment by tomtau** [comment]
```
4/5] Link (lld) libUE4Editor-CronosPlayUnreal.so
84
  ld.lld: error: cannot open /home/ue4/play-unreal-plugin/Output/HostProject/Plugins/CronosPlayUnreal/Source/ThirdParty/PlayQrcodeLibrary/Lib/Linux/libqrcodegen.a: No such file or directory
85
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by leejw51crypto** [comment]
added compiled qrcode for linux

**Comment by tomtau** [comment]
can this work without including the binary blob? Can the build.cs invoke the makefile before linking?

**Comment by leejw51crypto** [comment]
for thirdparty, binary should be used.
i tried source file, didn't linked successfully.

i'm fixing linux compile errors.

**Comment by tomtau** [comment]
how I can verify the uploaded binary is correct?

**Comment by leejw51crypto** [comment]
how about providing sha256sum for thirdparty binaries?

**Comment by tomtau** [comment]
> how about providing sha256sum for thirdparty binaries?

so for this code review, I should build all the binaries and check if the sha256sum matches the ones you uploaded?

**Comment by leejw51crypto** [comment]
nope, no need to recompile third-party binaries.
let's say one script to check sha256sum for third party binaries to ensure whether it's the same with first original one?

**Comment by leejw51crypto** [comment]
for qrcode, i'll check for compiling as third party library.

**Comment by leejw51crypto** [comment]
> can this work without including the binary blob? Can the build.cs invoke the makefile before linking?

this removes binary repo, no manual uploading.
only use PlayCppSdk , and ue plugin generated in github server

**Comment by leejw51crypto** [comment]
temporarily using play-cpp-sdk v0.0.6-alpha for test

**Review by tomtau — APPROVED** [review]

**Comment by damoncro** [comment]
CHANGELOG.md of play-cpp-sdk library is not updated.

**Comment by leejw51crypto** [comment]
i'll update changelog of play-cpp-sdk

**Pull request merged**